Controller/Annotations/QueryParam.php

```diff
@@ -17,12 +17,40 @@
  * Represents a parameter that must be present in GET data.
  *
  * @Annotation
+ * @NamedArgumentConstructor
  * @Target({"CLASS", "METHOD"})
  *
  * @author Alexander <iam.asm89@gmail.com>
  */
+#[\Attribute(\Attribute::IS_REPEATABLE | \Attribute::TARGET_CLASS | \Attribute::TARGET_METHOD)]
 class QueryParam extends AbstractScalarParam
 {
+    /**
+     * @param mixed $requirements
+     * @param mixed $default
+     */
+    public function __construct(
+        string $name = '',
```

Comment:
@GuilhemN I guess you refer to the BC layer for the `Route` annotation.

I'm not sure how such BC layer should look like here.
Before my changes the constructor of the annotations were just empty.

Since all newly added parameters have default values it is still possible to call `new QueryParam()` and the behavior of this call should not change.